hbase-common/pom.xml

```diff
@@ -110,6 +110,10 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-crypto</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.github.ben-manes.caffeine</groupId>
+      <artifactId>caffeine</artifactId>
+    </dependency>
```

Comment:
Why we need an extra library? Can't we use `google.common.cache`, just like the CodecPool from hadoop-commons?

Reply:
I have a future PR coming after this one where I'll want the highest-performance cache here possible, which is Caffeine. I thought since I'll be introducing that later I might as well put this cache on it now. The HBase project uses Caffeine already, although not in hbase-common.

Reply:
fwiw, if desired neither are necessary. The use-case appears to be an unbounded concurrent map with a fast computeIfAbsent.

ConcurrentHashMap tends to lock pessimistically which impacts reads. While improved in Java 9 to skip that if found on as the first item in the bin, it’s rare. Caffeine does a lock-free get and falls back to a compute. This tends to be much faster despite a little extra work. Here’s a [benchmark run](https://jmh.morethan.io/?gists=511298014cc5629cbc5e57f09fd4c430) on the CI.

Just an fyi if helpful information.

Reply:
The man himself! I appreciate the info!